docs/application.rst

```diff
@@ -1,80 +1,75 @@
-  - all the user-based stuff like `register_form`, `user`, `user_form`,
-    `forgotten_password_form`, `login`, `logout`, `forgotten_password`,
-    `password_reset`, `pw_reset` and `pw_reset_change` will most likely be
-    replaced by newer mechanisms in warehouse
-  - `openid_endpoint`, `openid_decide_post` could also be replaced by something
-    else.
-  - `home` is the old home page thing and completely unnecessary
-  - `index` is overwhelming given the number of projects now.
-  - `browse` and `search` are *probably* only referenced by internal links so
-    should be safe to nuke
-  - `submit_pkg_info` and `display_pkginfo` probably aren't used
-  - `submit_form` and `pkg_edit` will be changing anyway
-  - `files`, `urls`, `role`, `role_form` are old style and will be changing
-  - `list_classifiers` .. this might actually only be used by Richard :)
-  - `claim`, `openid`, `openid_return`, `dropid` are legacy openid login
-    support and will be changing
-  - `clear_auth` "clears" Basic Auth
-  - `addkey`, `delkey` will be changing if we even keep supporting ssh submit
-  - `verify` probably isn't actually used by anyone
-  - `lasthour` is a pubsubhubbub thing - does this even exist any longer?
-  - `json` is never used as a :action invocation, only ever /<package>/json
-  - `gae_file` I'm pretty sure this is not necessary
-  - `rss_regen` manually regens the RSS cached files, not needed
-  - `about` No longer needed.
-  - `delete_user` No longer needed.
-  - `exception` No longer needed.
+-  ``LICENSE``
+-  ``CONTRIBUTING.rst`` (the contribution guide)
+-  ``README.rst``
+-  ``requirements.txt`` for the Warehouse virtual environment
+-  ``Dockerfile``: creates the Docker containers that Warehouse runs in
+-  ``docker-compose.yml`` file configures Docker Compose
+-  ``setup.cfg`` for test configuration
+-  ``runtime.txt`` for Heroku
+-  ``Makefile``: commands to spin up Docker Compose and the Docker
+   containers
+-  files associated with Warehouse's front end, e.g.,
+   ``Gulpfile.babel.js``
 
-**will need to retain**
-  - `rss` and `packages_rss` will be in a bunch of peoples` RSS readers
-  - `doap` is most likely referred to
-  - `show_md5` ?
+Directories within the repository:
 
-**can be deprecated carefully**
-  - `submit`, `upload`, `doc_upload`, `file_upload`,
+::
+
+    bin/ - high-level scripts for Docker
+    dev/ - assets for developer environment
+    tests/ - tests
+    warehouse/ - code in modules
+        legacy/ - most of the read-only APIs implemented here
+        forklift/ - APIs for upload
+        accounts/ - user accounts
+        admin/ - application-administrator-specific
+        cache/ - caching
+        classifiers/ - frame trove classifiers
+        cli/ - entry scripts and [the interactive shell](https://warehouse.readthedocs.io/development/getting-started/#running-the-interactive-shell)
+        i18n/ - internationalization
+        locales/ - internationalization
+        manage/ - DB
```

Comment:
This is actually for everything a logged-in user can do, (i.e. manage account and owned projects)